{"dependencies": [{"name": "agno", "version": "1.7.5", "vulns": []}, {"name": "aiofiles", "version": "24.1.0", "vulns": []}, {"name": "aiohappyeyeballs", "version": "2.6.1", "vulns": []}, {"name": "aiohttp", "version": "3.12.14", "vulns": []}, {"name": "aiosignal", "version": "1.4.0", "vulns": []}, {"name": "alembic", "version": "1.16.4", "vulns": []}, {"name": "annotated-types", "version": "0.7.0", "vulns": []}, {"name": "anthropic", "version": "0.57.1", "vulns": []}, {"name": "anyio", "version": "4.9.0", "vulns": []}, {"name": "asyncer", "version": "0.0.8", "vulns": []}, {"name": "asyncpg", "version": "0.30.0", "vulns": []}, {"name": "attrs", "version": "25.3.0", "vulns": []}, {"name": "authlib", "version": "1.6.1", "vulns": []}, {"name": "automagik-hive", "skip_reason": "Dependency not found on PyPI and could not be audited: automagik-hive (0.1.2)"}, {"name": "backoff", "version": "2.2.1", "vulns": []}, {"name": "bandit", "version": "1.8.6", "vulns": []}, {"name": "boolean-py", "version": "5.0", "vulns": []}, {"name": "cachecontrol", "version": "0.14.3", "vulns": []}, {"name": "cachetools", "version": "5.5.2", "vulns": []}, {"name": "certifi", "version": "2025.7.14", "vulns": []}, {"name": "cffi", "version": "1.17.1", "vulns": []}, {"name": "charset-normalizer", "version": "3.4.2", "vulns": []}, {"name": "click", "version": "8.2.1", "vulns": []}, {"name": "cloudpickle", "version": "3.1.1", "vulns": []}, {"name": "cohere", "version": "5.16.1", "vulns": []}, {"name": "colorlog", "version": "6.9.0", "vulns": []}, {"name": "coolname", "version": "2.2.0", "vulns": []}, {"name": "coverage", "version": "7.9.2", "vulns": []}, {"name": "cryptography", "version": "45.0.5", "vulns": []}, {"name": "cyclonedx-python-lib", "version": "9.1.0", "vulns": []}, {"name": "datasets", "version": "4.0.0", "vulns": []}, {"name": "decorator", "version": "5.2.1", "vulns": []}, {"name": "defusedxml", "version": "0.7.1", "vulns": []}, {"name": "deprecated", "version": "1.2.18", "vulns": []}, {"name": "dill", "version": "0.3.8", "vulns": []}, {"name": "diskcache", "version": "5.6.3", "vulns": []}, {"name": "distro", "version": "1.9.0", "vulns": []}, {"name": "docstring-parser", "version": "0.16", "vulns": []}, {"name": "dparse", "version": "0.6.4", "vulns": []}, {"name": "dspy", "version": "2.6.27", "vulns": []}, {"name": "dspy-ai", "version": "2.6.27", "vulns": []}, {"name": "faker", "version": "37.4.2", "vulns": []}, {"name": "fastapi", "version": "0.116.1", "vulns": []}, {"name": "fastavro", "version": "1.11.1", "vulns": []}, {"name": "filelock", "version": "3.16.1", "vulns": []}, {"name": "frozenlist", "version": "1.7.0", "vulns": []}, {"name": "fsspec", "version": "2025.3.0", "vulns": []}, {"name": "gitdb", "version": "4.0.12", "vulns": []}, {"name": "gitpython", "version": "3.1.44", "vulns": []}, {"name": "google-ai-generativelanguage", "version": "0.6.15", "vulns": []}, {"name": "google-api-core", "version": "2.25.1", "vulns": []}, {"name": "google-api-python-client", "version": "2.176.0", "vulns": []}, {"name": "google-auth", "version": "2.40.3", "vulns": []}, {"name": "google-auth-httplib2", "version": "0.2.0", "vulns": []}, {"name": "google-genai", "version": "1.26.0", "vulns": []}, {"name": "google-generativeai", "version": "0.8.5", "vulns": []}, {"name": "googleapis-common-protos", "version": "1.70.0", "vulns": []}, {"name": "greenlet", "version": "3.2.3", "vulns": []}, {"name": "grpcio", "version": "1.73.1", "vulns": []}, {"name": "grpcio-status", "version": "1.71.2", "vulns": []}, {"name": "h11", "version": "0.16.0", "vulns": []}, {"name": "hf-xet", "version": "1.1.5", "vulns": []}, {"name": "httpcore", "version": "1.0.9", "vulns": []}, {"name": "httplib2", "version": "0.22.0", "vulns": []}, {"name": "httpx", "version": "0.28.1", "vulns": []}, {"name": "httpx-sse", "version": "0.4.0", "vulns": []}, {"name": "huggingface-hub", "version": "0.33.4", "vulns": []}, {"name": "idna", "version": "3.10", "vulns": []}, {"name": "importlib-metadata", "version": "8.7.0", "vulns": []}, {"name": "iniconfig", "version": "2.1.0", "vulns": []}, {"name": "jinja2", "version": "3.1.6", "vulns": []}, {"name": "jiter", "version": "0.10.0", "vulns": []}, {"name": "joblib", "version": "1.5.1", "vulns": []}, {"name": "json-repair", "version": "0.47.8", "vulns": []}, {"name": "jsonschema", "version": "4.24.1", "vulns": []}, {"name": "jsonschema-specifications", "version": "2025.4.1", "vulns": []}, {"name": "langwatch", "version": "0.2.9", "vulns": []}, {"name": "license-expression", "version": "30.4.4", "vulns": []}, {"name": "litellm", "version": "1.74.7", "vulns": []}, {"name": "loguru", "version": "0.7.3", "vulns": []}, {"name": "magicattr", "version": "0.1.6", "vulns": []}, {"name": "mako", "version": "1.3.10", "vulns": []}, {"name": "markdown", "version": "3.8.2", "vulns": []}, {"name": "markdown-it-py", "version": "3.0.0", "vulns": []}, {"name": "markupsafe", "version": "3.0.2", "vulns": []}, {"name": "marshmallow", "version": "4.0.0", "vulns": []}, {"name": "mcp", "version": "1.11.0", "vulns": []}, {"name": "mdurl", "version": "0.1.2", "vulns": []}, {"name": "msgpack", "version": "1.1.1", "vulns": []}, {"name": "multidict", "version": "6.6.3", "vulns": []}, {"name": "multiprocess", "version": "0.70.16", "vulns": []}, {"name": "mypy", "version": "1.17.0", "vulns": []}, {"name": "mypy-extensions", "version": "1.1.0", "vulns": []}, {"name": "nanoid", "version": "2.0.0", "vulns": []}, {"name": "nltk", "version": "3.9.1", "vulns": []}, {"name": "numpy", "version": "2.3.1", "vulns": []}, {"name": "openai", "version": "1.97.0", "vulns": []}, {"name": "openinference-instrumentation", "version": "0.1.35", "vulns": []}, {"name": "openinference-instrumentation-agno", "version": "0.1.9", "vulns": []}, {"name": "openinference-instrumentation-haystack", "version": "0.1.24", "vulns": []}, {"name": "openinference-instrumentation-langchain", "version": "0.1.47", "vulns": []}, {"name": "openinference-instrumentation-openai", "version": "0.1.30", "vulns": []}, {"name": "openinference-semantic-conventions", "version": "0.1.21", "vulns": []}, {"name": "opentelemetry-api", "version": "1.35.0", "vulns": []}, {"name": "opentelemetry-exporter-otlp-proto-common", "version": "1.35.0", "vulns": []}, {"name": "opentelemetry-exporter-otlp-proto-http", "version": "1.35.0", "vulns": []}, {"name": "opentelemetry-instrumentation", "version": "0.56b0", "vulns": []}, {"name": "opentelemetry-proto", "version": "1.35.0", "vulns": []}, {"name": "opentelemetry-sdk", "version": "1.35.0", "vulns": []}, {"name": "opentelemetry-semantic-conventions", "version": "0.56b0", "vulns": []}, {"name": "optuna", "version": "4.4.0", "vulns": []}, {"name": "packageurl-python", "version": "0.17.1", "vulns": []}, {"name": "packaging", "version": "25.0", "vulns": []}, {"name": "pandas", "version": "2.3.1", "vulns": []}, {"name": "pathlib", "version": "1.0.1", "vulns": []}, {"name": "pathspec", "version": "0.12.1", "vulns": []}, {"name": "pbr", "version": "6.1.1", "vulns": []}, {"name": "pgvector", "version": "0.4.1", "vulns": []}, {"name": "pip", "version": "25.1.1", "vulns": []}, {"name": "pip-api", "version": "0.0.34", "vulns": []}, {"name": "pip-audit", "version": "2.9.0", "vulns": []}, {"name": "pip-requirements-parser", "version": "32.0.1", "vulns": []}, {"name": "platformdirs", "version": "4.3.8", "vulns": []}, {"name": "pluggy", "version": "1.6.0", "vulns": []}, {"name": "propcache", "version": "0.3.2", "vulns": []}, {"name": "proto-plus", "version": "1.26.1", "vulns": []}, {"name": "protobuf", "version": "5.29.5", "vulns": []}, {"name": "psutil", "version": "6.1.1", "vulns": []}, {"name": "psycopg", "version": "3.2.9", "vulns": []}, {"name": "psycopg-binary", "version": "3.2.9", "vulns": []}, {"name": "psycopg-pool", "version": "3.2.6", "vulns": []}, {"name": "py", "version": "1.11.0", "vulns": [{"id": "PYSEC-2022-42969", "fix_versions": [], "aliases": ["GHSA-w596-4wvx-j9j6", "CVE-2022-42969"], "description": "The py library through 1.11.0 for Python allows remote attackers to conduct a ReDoS (Regular expression Denial of Service) attack via a Subversion repository with crafted info data, because the InfoSvnCommand argument is mishandled."}]}, {"name": "py-serializable", "version": "2.1.0", "vulns": []}, {"name": "pyarrow", "version": "21.0.0", "vulns": []}, {"name": "pyasn1", "version": "0.6.1", "vulns": []}, {"name": "pyasn1-modules", "version": "0.4.2", "vulns": []}, {"name": "pycparser", "version": "2.22", "vulns": []}, {"name": "pydantic", "version": "2.9.2", "vulns": []}, {"name": "pydantic-core", "version": "2.23.4", "vulns": []}, {"name": "pydantic-settings", "version": "2.10.1", "vulns": []}, {"name": "pygments", "version": "2.19.2", "vulns": []}, {"name": "pyparsing", "version": "3.2.3", "vulns": []}, {"name": "pytest", "version": "8.4.1", "vulns": []}, {"name": "pytest-asyncio", "version": "1.1.0", "vulns": []}, {"name": "pytest-cov", "version": "6.2.1", "vulns": []}, {"name": "pytest-mock", "version": "3.14.1", "vulns": []}, {"name": "python-dateutil", "version": "2.9.0.post0", "vulns": []}, {"name": "python-dotenv", "version": "1.1.1", "vulns": []}, {"name": "python-multipart", "version": "0.0.20", "vulns": []}, {"name": "pytz", "version": "2025.2", "vulns": []}, {"name": "pyyaml", "version": "6.0.2", "vulns": []}, {"name": "referencing", "version": "0.36.2", "vulns": []}, {"name": "regex", "version": "2024.11.6", "vulns": []}, {"name": "requests", "version": "2.32.4", "vulns": []}, {"name": "retry", "version": "0.9.2", "vulns": []}, {"name": "rich", "version": "14.0.0", "vulns": []}, {"name": "rpds-py", "version": "0.26.0", "vulns": []}, {"name": "rsa", "version": "4.9.1", "vulns": []}, {"name": "ruamel-yaml", "version": "0.18.14", "vulns": []}, {"name": "ruamel-yaml-clib", "version": "0.2.12", "vulns": []}, {"name": "ruff", "version": "0.12.3", "vulns": []}, {"name": "safety", "version": "3.6.0", "vulns": []}, {"name": "safety-schemas", "version": "0.0.14", "vulns": []}, {"name": "setuptools", "version": "80.9.0", "vulns": []}, {"name": "shellingham", "version": "1.5.4", "vulns": []}, {"name": "six", "version": "1.17.0", "vulns": []}, {"name": "smmap", "version": "5.0.2", "vulns": []}, {"name": "sniffio", "version": "1.3.1", "vulns": []}, {"name": "sortedcontainers", "version": "2.4.0", "vulns": []}, {"name": "sqlalchemy", "version": "2.0.41", "vulns": []}, {"name": "sse-starlette", "version": "2.4.1", "vulns": []}, {"name": "starlette", "version": "0.47.1", "vulns": [{"id": "GHSA-2c2j-9gv5-cj73", "fix_versions": ["0.47.2"], "aliases": ["CVE-2025-54121"], "description": "### Summary When parsing a multi-part form with large files (greater than the [default max spool size](https://github.com/encode/starlette/blob/fa5355442753f794965ae1af0f87f9fec1b9a3de/starlette/formparsers.py#L126)) `starlette` will block the main thread to roll the file over to disk. This blocks the event thread which means we can't accept new connections.  ### Details Please see this discussion for details: https://github.com/encode/starlette/discussions/2927#discussioncomment-13721403. In summary the following UploadFile code (copied from [here](https://github.com/encode/starlette/blob/fa5355442753f794965ae1af0f87f9fec1b9a3de/starlette/datastructures.py#L436C5-L447C14)) has a minor bug. Instead of just checking for `self._in_memory` we should also check if the additional bytes will cause a rollover.  ```python      @property     def _in_memory(self) -> bool:         # check for SpooledTemporaryFile._rolled         rolled_to_disk = getattr(self.file, \"_rolled\", True)         return not rolled_to_disk      async def write(self, data: bytes) -> None:         if self.size is not None:             self.size += len(data)          if self._in_memory:             self.file.write(data)         else:             await run_in_threadpool(self.file.write, data) ```  I have already created a PR which fixes the problem: https://github.com/encode/starlette/pull/2962   ### PoC See the discussion [here](https://github.com/encode/starlette/discussions/2927#discussioncomment-13721403) for steps on how to reproduce.  ### Impact To be honest, very low and not many users will be impacted. Parsing large forms is already CPU intensive so the additional IO block doesn't slow down `starlette` that much on systems with modern HDDs/SSDs. If someone is running on tape they might see a greater impact."}]}, {"name": "stevedore", "version": "5.4.1", "vulns": []}, {"name": "tenacity", "version": "8.5.0", "vulns": []}, {"name": "termcolor", "version": "3.1.0", "vulns": []}, {"name": "tiktoken", "version": "0.9.0", "vulns": []}, {"name": "tokenizers", "version": "0.21.2", "vulns": []}, {"name": "toml", "version": "0.10.2", "vulns": []}, {"name": "tomli", "version": "2.2.1", "vulns": []}, {"name": "tomlkit", "version": "0.13.3", "vulns": []}, {"name": "tqdm", "version": "4.67.1", "vulns": []}, {"name": "typer", "version": "0.16.0", "vulns": []}, {"name": "types-pyyaml", "version": "6.0.12.20250516", "vulns": []}, {"name": "types-requests", "version": "2.32.4.20250611", "vulns": []}, {"name": "typing-extensions", "version": "4.14.1", "vulns": []}, {"name": "typing-inspection", "version": "0.4.1", "vulns": []}, {"name": "tzdata", "version": "2025.2", "vulns": []}, {"name": "ujson", "version": "5.10.0", "vulns": []}, {"name": "uritemplate", "version": "4.2.0", "vulns": []}, {"name": "urllib3", "version": "2.5.0", "vulns": []}, {"name": "uvicorn", "version": "0.35.0", "vulns": []}, {"name": "watchdog", "version": "6.0.0", "vulns": []}, {"name": "websockets", "version": "15.0.1", "vulns": []}, {"name": "wrapt", "version": "1.17.2", "vulns": []}, {"name": "xxhash", "version": "3.5.0", "vulns": []}, {"name": "yarl", "version": "1.20.1", "vulns": []}, {"name": "zipp", "version": "3.23.0", "vulns": []}], "fixes": []}
